build

```diff
@@ -1,54 +1,63 @@
+#--------------------------EDIT FROM HERE-------------------------------
+#  EDIT IMP VARIABLES AND COLLECT CCACHE BEFORE BUILD ELSE CI TIMEOUT
 
-# Normal build steps
-. build/envsetup.sh
-lunch aosp_mido-user
-export CCACHE_DIR=/tmp/ccache
-export CCACHE_EXEC=$(which ccache)
-export USE_CCACHE=1
-ccache -M 20G # It took only 6.4GB for mido
-ccache -o compression=true # Will save times and data to download and upload ccache, also negligible performance issue
-ccache -z # Clear old stats, so monitor script will provide real ccache statistics
-
-# Next 8 lines should be run first to collect ccache and then upload, after doning it 1 or 2 times, our ccache will help to build without these 8 lines
-#make api-stubs-docs || echo no problem, we need ccache
-#make hiddenapi-lists-docs || echo no problem, we need ccache
-#make system-api-stubs-docs || echo no problem we need ccache
-#make test-api-stubs-docs || echo no problem, we need ccache
-#make aex -j10 & # dont remove that '&'
-#sleep 85m
-#kill %1
-#ccache -s
-#and dont use below codes for first 1 or 2 times, to get ccache uploaded,
-
-
-# upload function for uploading rom zip file! I don't want unwanted builds in my google drive haha!
+export botapi="$mybot"  # add your own bot api                      
+export id="$myid"                                 # Your telegram id check in sangmatabot@tg
+export build_type="final"                         # for {final} build use final or use {ccache} to collect ccache
+export ccache_size="12G"                          # dont use above 10-15 gb dont abuse system
+export ccachedir="/tmp/ccache"                    # ccache dir dont edit if u dunno what ur doing
+export lunchcommand="lunch aosp_mido-userdebug"   # edit as per stated on manifest
+export buildcommand="make -j10 bacon"             # make command as per manifest
+
+#--------------------------EDIT UPTO HERE-------------------------------
+
+#env variables & functions
 up(){
-	curl --upload-file $1 https://transfer.sh/$(basename $1); echo
-	# 14 days, 10 GB limit
-}
+	curl --upload-file "$1" https://transfer.sh/"$(basename "$1")"; echo
+}                                                 # 14 days, 10 GB limit
 
 tg(){
-	bot_api=1692865707:AAHvKZogI5sUGtqjd2sZvNl8tib0Q_xcYkY # Your tg bot api, dont use my one haha
-	your_telegram_id=$1 # No need to touch 
-	msg=$2 # No need to touch
+	bot_api="$botapi"                              # Your tg bot api, dont use my one haha
+	your_telegram_id=$1                           # No need to edit
+	msg=$2                                        # No need to edit
 	curl -s "https://api.telegram.org/bot${bot_api}/sendmessage" --data "text=$msg&chat_id=${your_telegram_id}"
 }
+send_shell=$(tmate -S /tmp/tmate.sock display -p '#{tmate_ssh}') && export send_shell && export SELINUX_IGNORE_NEVERALLOWS=true
+export CCACHE_DIR="$ccachedir" && temp69="CCACHE_EXEC=$(which ccache)" && export temp69 && export USE_CCACHE=1
```

Comment:
its unnecessary to exporting same things twice! also if things can be done without export! why its even need to be exported???

Reply:
yes i agree that's my fault actually those export codes i had used form shellcheck.net website while verifying code